contracts/src/v0.8/automation/dev/test/AutomationRegistry2_3.t.sol

```diff
@@ -948,4 +948,25 @@ contract BillingOverrides is SetUp {
     emit BillingConfigOverrideRemoved(linkUpkeepID);
     registry.removeBillingOverrides(linkUpkeepID);
   }
+
+  function test_Happy_MaxGasPayment_WithBillingOverrides() public {
+    uint96 maxPayment1 = registry.getMaxPaymentForGas(linkUpkeepID, 0, 5_000_000, address(linkToken));
+    assertEq(maxPayment1, 2107776000000000000);
+
+    // Double the two billing values
+    AutomationRegistryBase2_3.BillingOverrides memory billingOverrides = AutomationRegistryBase2_3.BillingOverrides({
+      gasFeePPB: DEFAULT_GAS_FEE_PPB * 2,
+      flatFeeMilliCents: DEFAULT_FLAT_FEE_MILLI_CENTS * 2
+    });
+
+    vm.startPrank(PRIVILEGE_MANAGER);
+    registry.setBillingOverrides(linkUpkeepID, billingOverrides);
+
+    // maxPayment2 should be different than maxPayment1 after the overrides
+    // The 2 numbers should follow this: maxPayment2 - maxPayment1 == 2 * recepit.premium
+    // We use hardcoded values since we couldn't get the receipt.premium value
+    // If we ever change the math in _calculatePaymentAmount(), we should update this test
+    uint96 maxPayment2 = registry.getMaxPaymentForGas(linkUpkeepID, 0, 5_000_000, address(linkToken));
+    assertEq(maxPayment2, 2128776000000000000);
```

Comment:
if I can make one small request, could we write the test so that we expect `maxPayment2 > maxPayment1`? would that work? I think if we do a "hard equals", then we might have to update the test whenever we tweak something in the billing funciton.